inst/templates/circleci-config.yml

```diff
@@ -0,0 +1,25 @@
+version: 2
+jobs:
+  build:
+    docker:
+      - image: {{{ image }}}
+    steps:
+      - restore_cache:
+          keys:
+            - cache
+      - checkout
+      - run:
+          name: Install package dependencies
+          command: R -e "devtools::install_deps(dependencies = TRUE)"
+      - run:
+          name: Build package
+          command: R CMD build .
+      - run:
+          name: Check package
+          command: R CMD check --as-cran --no-manual *tar.gz
+      - store_artifacts:
+          path: {{{ package }}}.Rcheck/
+      - save_cache:
+          key: cache
+          paths:
+            - "/usr/local/lib/R/site-library"
```

Comment:
At least the rocker containers set the `R_LIBS_USER` environment variable (to this location), maybe this should be $R_LIBS_USER ? Although I am not sure if circleci will expand the environment variable here...

Reply:
Also IIRC because this is the default library location for the verse docker container, the CircleCI cache will contain all of the package already installed in the library. It might be better to do something like `R_LIBS_USER=/foo/bar:${R_LIBS_USER}` and setup new library that is only dependencies installed by CircleCI.

Reply:
This might be possible using [parameters](https://circleci.com/docs/2.0/reusing-config/#using-the-parameters-declaration).

But I do think that most users will use a rocker image, and any user that can create an R environment from a more minimal Docker image (e.g. a standard Ubuntu image) should be able to handle updating the library path.

Reply:
`R_LIBS_USER` has to be set in `.Renviron`, right? Do you want me to add a step to create this file?

Reply:
No, it is just a normal shell environment variable, it can be set like you would normally in the CI.

Reply:
Thanks for the confirmation. That's what I had thought, but I'm having trouble getting the rocker image to recognize `R_LIBS_USER`, even after creating that directory. It works on my local Ubuntu, so I'm not sure what is going wrong. Maybe this discussion is related: https://github.com/rocker-org/rocker-versioned/issues/108#issuecomment-445003537 I'll keep investigating.

Reply:
Starting a Docker container with `docker run --rm -it rocker/verse:3.5.2 bash`

```
export R_LIBS_USER=~/R/Library
echo $R_LIBS_USER
## /root/R/Library/
mkdir -p $R_LIBS_USER
Rscript -e 'Sys.getenv("R_LIBS_USER")'
## [1] "/usr/local/lib/R/site-library"
```

Reply:
OK. I confirmed that the problem is that the rocker image is setting `R_LIBS_USER` in a system `Renviron` ([line](https://github.com/rocker-org/rocker-versioned/blob/9ad0922a7cb6932dfacc5ecc387048a1e1cd6e59/r-ver/devel/Dockerfile#L115)). When I delete that line, then it respects the setting of the environment variable.

Reply:
@jimhester Since setting `R_LIBS_USER` as a standard environment variable didn't work in the rocker image, I tried defining it in `.Renviron`. This was at least recognized by R, but it was added to the *end* of the of the library path.

```
mkdir -p ~/R/Library
echo R_LIBS_USER=~/R/Library > ~/.Renviron
Rscript -e '.libPaths()'
## [1] "/usr/local/lib/R/site-library" "/usr/local/lib/R/library"     
## [3] "/root/R/Library"              
```

Thus I then tried using `.Rprofile`. I've had to resort to this hack before in order to configure R properly on my HPC. Note that only the call to `.libPaths()` is actually required. Including the call to `Sys.setenv()` is more for an end user trying to understand how the library paths were setup.

```
mkdir -p ~/R/Library
echo 'Sys.setenv(R_LIBS_USER = "~/R/Library")' > ~/.Rprofile
echo '.libPaths("~/R/Library")' >> ~/.Rprofile
cat ~/.Rprofile 
## Sys.setenv(R_LIBS_USER = "~/R/Library")
## .libPaths("~/R/Library")
Rscript -e '.libPaths()'
## [1] "/root/R/Library"               "/usr/local/lib/R/site-library"
## [3] "/usr/local/lib/R/library" 
```

Another alternative would be to set `lib = "~/R/Library"` when calling `install.packages()` and `remotes::install_deps()`.

Thoughts on which solution I should implement?

Reply:
The problem is the rocker container is also setting `R_LIBS` and `R_LIBS` takes precedence over `R_LIBS_USER`, so you have to use that in this case.

Reply:
Setting `R_LIBS` worked. Thanks! By setting `R_LIBS=~/R/library` and creating this directory, `.libPaths()` returns the desired order:

```
export R_LIBS=~/R/library
mkdir -p ~/R/library
Rscript -e '.libPaths()'
## [1] "/root/R/library"               "/usr/local/lib/R/site-library"
## [3] "/usr/local/lib/R/library" 
```